x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/api/connector/generate_connector_api_key_api_logic.ts

```diff
@@ -15,9 +15,23 @@ export interface ApiKey {
   name: string;
 }
 
-export const generateApiKey = async ({ indexName }: { indexName: string }) => {
+export const generateApiKey = async ({
+  indexName,
+  isNative,
+  secretId,
+}: {
+  indexName: string;
+  isNative: boolean;
+  secretId: string | null;
+}) => {
   const route = `/internal/enterprise_search/indices/${indexName}/api_key`;
-  return await HttpLogic.values.http.post<ApiKey>(route);
+  const params = {
+    isNative,
```

Comment:
nitpick: We usually want to use snake_case instead of camelCase for params and other variables in APIs.